ui/middleware.ts

```diff
@@ -43,22 +43,33 @@ const protectedPathnameRegex = RegExp(
 );
 
 export default async function middleware(req: NextRequest) {
-    /*
-     * Next.js middleware doesn't support reading files, so here we make a (cached)
-     * call to the /config endpoint within the same application :(
-     */
-  let oidcEnabled = await fetch(`http://127.0.0.1:${process.env.PORT}/config`, { cache: "force-cache" })
-    .then(cfg => cfg.json())
-    .then(cfg => cfg["oidc"]);
+  /*
+   * Next.js middleware doesn't support reading files, so here we make a (cached)
+   * call to the /config endpoint within the same application :(
+   */
+  let oidcEnabled = await fetch(`http://127.0.0.1:${process.env.PORT}/config`, {
+    cache: "force-cache",
+  })
+    .then((cfg) => cfg.json())
+    .then((cfg) => cfg["oidc"]);
+
+  const searchParams = req.nextUrl.searchParams;
 
   const requestPath = req.nextUrl.pathname;
-  const isPublicPage = !oidcEnabled && publicPathnameRegex.test(requestPath);
-  const isProtectedPage = oidcEnabled || protectedPathnameRegex.test(requestPath);
+  const isSchemaPublic =
+    requestPath === "/schema" &&
+    searchParams.has("content") &&
+    searchParams.has("schemaname");
+
+  const isPublicPage =
+    (!oidcEnabled && publicPathnameRegex.test(requestPath)) || isSchemaPublic;
```

Comment:
Making schemas public won't work without API changes. When OIDC is enabled, the entire `/api` path will be protected so these requests will return a 401. Can the schemas still be under `(authorized)` somehow? Is it conflicting with some other path?

Reply:
I understand the concern about the /api path being protected when OIDC is enabled. However, I am passing anonymous: true in the headers by using await getHeaders(true). This should allow the request to bypass authentication, similar to how Kafka logging is handled.

Reply:
Ah okay, I see the problem

Reply:
I have updated the PR to support OIDC, hope this will fix the issue